docs/source/api/python/udf.rst

```diff
@@ -46,4 +79,10 @@ The class that can represent a batch of nodes.
     NodeBatch.mailbox
     NodeBatch.nodes
     NodeBatch.batch_size
-    NodeBatch.__len__
+
+Degree Bucketing
+----------------
+
+DGL employs a degree-bucketing mechanism for message passing with UDFs. It groups nodes with
+a same in-degree and invokes message passing for each group of nodes. As a result, one shall
+not make any assumptions about the batch size of :class:`~dgl.udf.NodeBatch` instances.
```

Comment:
"DGL stacks the incoming messages of each node along the second dimension, ordering them by edge ID."

TreeLSTM depends on this behavior.  However, the ordering is not preserved in 0.5.1.  This is a bug (#2175).